scripts/stampy/upload-stampy-signed.js

```diff
@@ -0,0 +1,28 @@
+const fs = require('fs')
+const exec = require('child_process').exec
+
+fs.readdirSync('.')
+  .filter(f => f.startsWith('heroku-v') && f.endsWith('.exe'))
+  .map(f => ({
+    filename: f,
+    cli: f.split('-')[0],
+    // do it from the end to account for the possibility of the version having a hyphen in it like 2.2.2-beta.0
+    sha: f.split('-').at(-2),
```

Comment:
Would the [semver functions](https://github.com/npm/node-semver?tab=readme-ov-file#functions) help to parse out the version more easily?

Reply:
@justinwilaby I did not see anything there that would extract the version from a string such as `heroku-v8.5.6-6181927-x64`. Maybe I'm missing something? If you know of a function that would help, I would be happy to implement it. This is the code that the Salesforce CLI uses to extract the version from their Stampy signed files.

Reply:
I agree that those functions won't help for this exact problem, but they should allow us to delete a significant amount of logic elsewhere. Thanks @justinwilaby !

Reply:
@k80bowman - Did you look at the `parse()` function? 
> parse(v): Attempt to parse a string as a semantic version, returning either a SemVer object or null.

Reply:
That's too bad. Yeah, I looked at that function, too, I wish it did parse a string like that. It would make things much more elegant.